File: server/src/main/java/com/glucocare/server/feature/mission/dto/DailyMissionResponse.java
package com.glucocare.server.feature.mission.dto;

import com.glucocare.server.feature.mission.domain.MemberDailyMission;
import com.glucocare.server.feature.mission.domain.MissionType;

import java.time.LocalDate;

public record DailyMissionResponse(
        Long id,
        String title,
        String description,
        MissionType missionType,
        Double threshold,
        Double currentValue,
        Boolean canComplete,
        Boolean isCompleted,
        Boolean isFailed,
        Long rewardPoint,
        LocalDate missionDate
) {
    public static DailyMissionResponse from(MemberDailyMission dailyMission, Boolean canComplete, Double currentValue) {
        var mission = dailyMission.getMission();
        return new DailyMissionResponse(dailyMission.getId(),
                                        mission.getTitle(),
                                        mission.getDescription(),
                                        mission.getMissionType(),
                                        mission.getThreshold(),
                                        currentValue,
                                        canComplete,
                                        dailyMission.getIsCompleted(),
                                        dailyMission.getIsFailed(),
                                        dailyMission.getMission()
                                                    .getRewardPoint(),
                                        dailyMission.getDate());
    }
}

File: app/lib/features/mission/data/models/daily_mission_response.dart
import 'package:json_annotation/json_annotation.dart';

import 'mission_type.dart';

part 'daily_mission_response.g.dart';

@JsonSerializable()
class DailyMissionResponse {
  final int id;
  final String title;
  final String description;
  final MissionType missionType;
  final double threshold;
  final double currentValue;
  final bool canComplete;
  final bool isCompleted;
  final bool isFailed;
  final int rewardPoint;
  final DateTime missionDate;

  DailyMissionResponse({
    required this.id,
    required this.title,
    required this.description,
    required this.missionType,
    required this.threshold,
    required this.currentValue,
    required this.canComplete,
    required this.isCompleted,
    required this.isFailed,
    required this.rewardPoint,
    required this.missionDate,
  });

  factory DailyMissionResponse.fromJson(Map<String, dynamic> json) => _$DailyMissionResponseFromJson(json);

  Map<String, dynamic> toJson() => _$DailyMissionResponseToJson(this);

  @override
  String toString() {
    return 'DailyMissionResponse{id: $id, title: $title, description: $description, missionType: $missionType, threshold: $threshold, currentValue: $currentValue, canComplete: $canComplete, isCompleted: $isCompleted, isFailed: $isFailed, rewardPoint: $rewardPoint, missionDate: $missionDate}';
  }
}
